pkg/image/filefmt.go

```diff
@@ -0,0 +1,84 @@
+package image
+
+import (
+	"bytes"
+	"encoding/hex"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/golang/glog"
+	. "kubevirt.io/containerized-data-importer/pkg/common"
+)
+
+// Size of buffer used to read file headers.
+// Note: this is the size of tar's header. If a larger number is used the tar unarchive operation
+//   creates the destination file too large, by the difference between this const and 512.
+const MaxExpectedHdrSize = 512
+
+// key is file format, eg. "gz" or "tar", value is metadata describing the layout for this hdr
+type Headers map[string]Header
+
+var KnownHeaders = Headers{
```

Comment:
Let's not export `KnownHeaders` and have `CopyKnownHdrs()` act as a constructor for `Headers`